{
    "items": [
        {
            "tags": [
                "angular5",
                "webpack-dev-server",
                "monaco-editor"
            ],
            "owner": {
                "reputation": 692,
                "user_id": 1262566,
                "user_type": "registered",
                "accept_rate": 96,
                "profile_image": "https://i.stack.imgur.com/RF0Wv.png?s=128&g=1",
                "display_name": "Devid",
                "link": "https://stackoverflow.com/users/1262566/devid"
            },
            "is_answered": false,
            "view_count": 57,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1526463650,
            "creation_date": 1521639005,
            "last_edit_date": 1521641513,
            "question_id": 49407686,
            "body_markdown": "I am using the [ngx-monaco-editor][1] component together with Angular 5. \r\nThe monaco-editor is loading fine on localhost, but not on my dev-server. \r\n\r\n&gt; Loading failed for the script with source\r\n&gt; “https://se-dnexec.cic.muc/assets/monaco/vs/loader.js”.\r\n\r\nThis is my webpack.common configuration (under plugins):\r\n\r\n     new WriteFilePlugin(),\r\n     new CopyWebpackPlugin([\r\n        {\r\n          from: &#39;src/assets&#39;,\r\n          to: &#39;assets&#39;\r\n        },\r\n        {\r\n          from: &#39;src/meta&#39;\r\n        },\r\n        {\r\n          from: &#39;node_modules/ngx-monaco-editor/assets/monaco&#39;,\r\n          to: &#39;assets/monaco/&#39;,\r\n        }\r\n      ],\r\n        isProd ? {\r\n          ignore: [&#39;mock-data/**/*&#39;]\r\n        } : undefined\r\n      ),\r\n\r\n\r\nIn webpack.dev I am using the configuration from webpack.common.\r\n\r\nWhen I run the application from localhost under assets I can see this.\r\n[![localhost_monaco][2]][2]\r\n\r\nBut when running from dev-server I see that the monaco folder is missing:\r\n\r\n[![devserver_monaco][3]][3]\r\n\r\n\r\nI am using:\r\n\r\n - copy-webpack-plugin: &quot;^4.5.1&quot;,\r\n - write-file-webpack-plugin: &quot;^4.2.0&quot;\r\n - webpack: &quot;^3.11.0&quot;,\r\n - webpack-dev-server: &quot;~2.7.1&quot;,\r\n\r\nI would expect that when I run the dev-server that under assets folder the monaco folder is also copied, but for some reason that is not the case. Maybe someone experienced the same problem and can help me. (If further info is needed I can provide it).\r\n\r\nI tried the following solution: [Does not copy files to actual output folder when webpack-dev-server is used][4]\r\n\r\n\r\n  [1]: https://github.com/atularen/ngx-monaco-editor\r\n  [2]: https://i.stack.imgur.com/qDlNA.png\r\n  [3]: https://i.stack.imgur.com/PwtYV.png\r\n  [4]: https://github.com/webpack-contrib/copy-webpack-plugin/issues/29#issuecomment-348031891",
            "link": "https://stackoverflow.com/questions/49407686/monaco-editor-and-webpack-dev-server-not-working",
            "title": "Monaco-editor and webpack dev server not working",
            "body": "<p>I am using the <a href=\"https://github.com/atularen/ngx-monaco-editor\" rel=\"nofollow noreferrer\">ngx-monaco-editor</a> component together with Angular 5. \nThe monaco-editor is loading fine on localhost, but not on my dev-server. </p>\n\n<blockquote>\n  <p>Loading failed for the script with source\n  “<a href=\"https://se-dnexec.cic.muc/assets/monaco/vs/loader.js\" rel=\"nofollow noreferrer\">https://se-dnexec.cic.muc/assets/monaco/vs/loader.js</a>”.</p>\n</blockquote>\n\n<p>This is my webpack.common configuration (under plugins):</p>\n\n<pre><code> new WriteFilePlugin(),\n new CopyWebpackPlugin([\n    {\n      from: 'src/assets',\n      to: 'assets'\n    },\n    {\n      from: 'src/meta'\n    },\n    {\n      from: 'node_modules/ngx-monaco-editor/assets/monaco',\n      to: 'assets/monaco/',\n    }\n  ],\n    isProd ? {\n      ignore: ['mock-data/**/*']\n    } : undefined\n  ),\n</code></pre>\n\n<p>In webpack.dev I am using the configuration from webpack.common.</p>\n\n<p>When I run the application from localhost under assets I can see this.\n<a href=\"https://i.stack.imgur.com/qDlNA.png\" rel=\"nofollow noreferrer\"><img src=\"https://i.stack.imgur.com/qDlNA.png\" alt=\"localhost_monaco\"></a></p>\n\n<p>But when running from dev-server I see that the monaco folder is missing:</p>\n\n<p><a href=\"https://i.stack.imgur.com/PwtYV.png\" rel=\"nofollow noreferrer\"><img src=\"https://i.stack.imgur.com/PwtYV.png\" alt=\"devserver_monaco\"></a></p>\n\n<p>I am using:</p>\n\n<ul>\n<li>copy-webpack-plugin: \"^4.5.1\",</li>\n<li>write-file-webpack-plugin: \"^4.2.0\"</li>\n<li>webpack: \"^3.11.0\",</li>\n<li>webpack-dev-server: \"~2.7.1\",</li>\n</ul>\n\n<p>I would expect that when I run the dev-server that under assets folder the monaco folder is also copied, but for some reason that is not the case. Maybe someone experienced the same problem and can help me. (If further info is needed I can provide it).</p>\n\n<p>I tried the following solution: <a href=\"https://github.com/webpack-contrib/copy-webpack-plugin/issues/29#issuecomment-348031891\" rel=\"nofollow noreferrer\">Does not copy files to actual output folder when webpack-dev-server is used</a></p>\n"
        },
        {
            "tags": [
                "python",
                "parsing"
            ],
            "owner": {
                "reputation": 2304,
                "user_id": 1574104,
                "user_type": "registered",
                "accept_rate": 73,
                "profile_image": "https://www.gravatar.com/avatar/685bf9f1f469f90c37687d7e715ed964?s=128&d=identicon&r=PG",
                "display_name": "WebQube",
                "link": "https://stackoverflow.com/users/1574104/webqube"
            },
            "is_answered": true,
            "view_count": 15,
            "closed_date": 1526463609,
            "accepted_answer_id": 50367387,
            "answer_count": 2,
            "score": 1,
            "last_activity_date": 1526463644,
            "creation_date": 1526460867,
            "question_id": 50366401,
            "body_markdown": "I&#39;m using my parser like so\r\n\r\n     parser.add_argument(&#39;owner_type&#39;, type=int, required=False, location=location)\r\n\r\nI want to be able to send both `int` and `str` in that field `owner_type`. \r\nis there a way to do that? \r\n\r\nDidn&#39;t find anything in the docs. ",
            "link": "https://stackoverflow.com/questions/50366401/parser-add-argument-support-multiple-types",
            "closed_reason": "duplicate",
            "title": "parser.add_argument support multiple types",
            "body": "<p>I'm using my parser like so</p>\n\n<pre><code> parser.add_argument('owner_type', type=int, required=False, location=location)\n</code></pre>\n\n<p>I want to be able to send both <code>int</code> and <code>str</code> in that field <code>owner_type</code>. \nis there a way to do that? </p>\n\n<p>Didn't find anything in the docs. </p>\n"
        },
        {
            "tags": [
                "docker",
                "docker-machine",
                "etcd",
                "docker-swarm"
            ],
            "owner": {
                "reputation": 457,
                "user_id": 4629076,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/43fbd638b147aa4698ee7a032a293a66?s=128&d=identicon&r=PG",
                "display_name": "JorelC",
                "link": "https://stackoverflow.com/users/4629076/jorelc"
            },
            "is_answered": false,
            "view_count": 339,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1526463640,
            "creation_date": 1450733684,
            "last_edit_date": 1450735940,
            "question_id": 34404707,
            "body_markdown": "I have successfully running an UPC cluster with 3 ubuntu instances. \r\n\r\nI want to access the docker swarm environment, either *docker-machine*, or *docker swarm manage*, but it has not been possible. The closest I&#39;ve been has been adding machines via *docker-machine* with generic driver, but when I  try to add this using the *--swarm* parameter, this tells me that the machine does not belong to a Docker Swarm Cluster.\r\n\r\nWhat I want to do is to execute commands within the Docker Swarm Cluster created with **UPC (Universal Control Panel)**.",
            "link": "https://stackoverflow.com/questions/34404707/manage-docker-environment-for-swarm-running-cluster-with-upc",
            "title": "Manage docker environment for Swarm running cluster with UPC",
            "body": "<p>I have successfully running an UPC cluster with 3 ubuntu instances. </p>\n\n<p>I want to access the docker swarm environment, either <em>docker-machine</em>, or <em>docker swarm manage</em>, but it has not been possible. The closest I've been has been adding machines via <em>docker-machine</em> with generic driver, but when I  try to add this using the <em>--swarm</em> parameter, this tells me that the machine does not belong to a Docker Swarm Cluster.</p>\n\n<p>What I want to do is to execute commands within the Docker Swarm Cluster created with <strong>UPC (Universal Control Panel)</strong>.</p>\n"
        },
        {
            "tags": [
                "java",
                "maven",
                "maven-2",
                "maven-3"
            ],
            "owner": {
                "reputation": 344,
                "user_id": 4091182,
                "user_type": "registered",
                "accept_rate": 69,
                "profile_image": "https://i.stack.imgur.com/JH1Pj.png?s=128&g=1",
                "display_name": "Ildelian",
                "link": "https://stackoverflow.com/users/4091182/ildelian"
            },
            "is_answered": true,
            "view_count": 14,
            "accepted_answer_id": 50348487,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1526463631,
            "creation_date": 1526381263,
            "last_edit_date": 1526463631,
            "question_id": 50348296,
            "body_markdown": "I have a Maven Java project, that uses maven-resources-plugin version 2.3, I am trying to migrate to a newer maven-resources-plugin version ,but when I put a versior later than 2.3, the propertys are not replaced.\r\n\r\nExamples:\r\nWith 2.3 version, compiled application.properties:\r\n\r\n    server.contextPath=/rsc/\r\n    server.port=8082\r\n    trac.datasource.url=jdbc:mariadb://localhost:3306/RSC\r\n\r\nWith version 2.4 or later:\r\n\r\n    server.contextPath=/rsc/\r\n    server.port=${prop.server.port}\r\n    trac.datasource.url=${prop.trac.datasource.url}\r\n\r\nAl the props are on the POM, on a Profile:\r\n\r\n    &lt;profiles&gt;\r\n    \t\t&lt;!-- Develop Profile --&gt;\r\n    \t\t&lt;profile&gt;\r\n    \t\t\t&lt;id&gt;develop&lt;/id&gt;\r\n    \t\t\t&lt;activation&gt;\r\n    \t\t\t\t&lt;activeByDefault&gt;true&lt;/activeByDefault&gt;\r\n    \t\t\t\t&lt;property&gt;\r\n    \t\t\t\t\t&lt;name&gt;environment.type&lt;/name&gt;\r\n    \t\t\t\t\t&lt;value&gt;develop&lt;/value&gt;\r\n    \t\t\t\t&lt;/property&gt;\r\n    \t\t\t&lt;/activation&gt;\r\n    \t\t\t&lt;properties&gt;\r\n    \t\t\t\t&lt;prop.server.port&gt;8082&lt;/prop.server.port&gt;\r\n    \t\t\t\t&lt;prop.trac.datasource.url&gt;jdbc:mariadb://localhost:3306/RSC&lt;/prop.trac.datasource.url&gt;\r\n    \t\t\t&lt;/properties&gt;\r\n    \t\t&lt;/profile&gt;\r\n    \t&lt;/profiles&gt;\r\n\r\nOn version 2.4 and later, I get this error:\r\n\r\n    Could not resolve placeholder &#39;prop.trac.datasource.url&#39; in value &quot;${prop.trac.datasource.url}&quot;\r\n\r\nBecause the property is not replaced with the value. Any ideas? I do not see where the problem is.",
            "link": "https://stackoverflow.com/questions/50348296/maven-resource-plugin-migration",
            "title": "Maven Resource Plugin Migration",
            "body": "<p>I have a Maven Java project, that uses maven-resources-plugin version 2.3, I am trying to migrate to a newer maven-resources-plugin version ,but when I put a versior later than 2.3, the propertys are not replaced.</p>\n\n<p>Examples:\nWith 2.3 version, compiled application.properties:</p>\n\n<pre><code>server.contextPath=/rsc/\nserver.port=8082\ntrac.datasource.url=jdbc:mariadb://localhost:3306/RSC\n</code></pre>\n\n<p>With version 2.4 or later:</p>\n\n<pre><code>server.contextPath=/rsc/\nserver.port=${prop.server.port}\ntrac.datasource.url=${prop.trac.datasource.url}\n</code></pre>\n\n<p>Al the props are on the POM, on a Profile:</p>\n\n<pre><code>&lt;profiles&gt;\n        &lt;!-- Develop Profile --&gt;\n        &lt;profile&gt;\n            &lt;id&gt;develop&lt;/id&gt;\n            &lt;activation&gt;\n                &lt;activeByDefault&gt;true&lt;/activeByDefault&gt;\n                &lt;property&gt;\n                    &lt;name&gt;environment.type&lt;/name&gt;\n                    &lt;value&gt;develop&lt;/value&gt;\n                &lt;/property&gt;\n            &lt;/activation&gt;\n            &lt;properties&gt;\n                &lt;prop.server.port&gt;8082&lt;/prop.server.port&gt;\n                &lt;prop.trac.datasource.url&gt;jdbc:mariadb://localhost:3306/RSC&lt;/prop.trac.datasource.url&gt;\n            &lt;/properties&gt;\n        &lt;/profile&gt;\n    &lt;/profiles&gt;\n</code></pre>\n\n<p>On version 2.4 and later, I get this error:</p>\n\n<pre><code>Could not resolve placeholder 'prop.trac.datasource.url' in value \"${prop.trac.datasource.url}\"\n</code></pre>\n\n<p>Because the property is not replaced with the value. Any ideas? I do not see where the problem is.</p>\n"
        },
        {
            "tags": [
                "java-8"
            ],
            "owner": {
                "reputation": 50,
                "user_id": 8536172,
                "user_type": "registered",
                "accept_rate": 100,
                "profile_image": "https://www.gravatar.com/avatar/9ca16a1623ff0f0eb97ec067994c0e20?s=128&d=identicon&r=PG&f=1",
                "display_name": "Arnab",
                "link": "https://stackoverflow.com/users/8536172/arnab"
            },
            "is_answered": true,
            "view_count": 32,
            "accepted_answer_id": 50366389,
            "answer_count": 3,
            "score": 1,
            "last_activity_date": 1526463624,
            "creation_date": 1526456632,
            "last_edit_date": 1526458209,
            "question_id": 50365049,
            "body_markdown": "I am trying to parse a date string returned to me from DB to LocalDate using DateTimeFormatter. I am receiving the below exception.\r\n\r\n    String date = &quot;2018-05-16 03:39:13.0&quot;;\r\n    \r\n    DateTimeFormatter formatter = DateTimeFormatter.ofPattern(&quot;yyyy-MM-dd HH:mm:ss.SSS&quot;);\r\n    LocalDateTime localDate= LocalDateTime.parse(date , formatter);\r\n\r\n___\r\n    java.time.format.DateTimeParseException: Text &#39;2018-05-16 03:39:13.0&#39; could not be parsed at index 20\r\n            at java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:1949) [rt.jar:1.8.0_171]\r\n            at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1851) [rt.jar:1.8.0_171]\r\n            at java.time.LocalDateTime.parse(LocalDateTime.java:492) [rt.jar:1.8.0_171]\r\n\r\n\r\nHowever the below code is working.\r\n\r\n    String date = &quot;2018-05-16 03:39:13.0&quot;;\r\n    SimpleDateFormat formatter = new SimpleDateFormat(&quot;yyyy-MM-dd HH:mm:ss.SSS&quot;);\r\n    Date newDate= formatter.parse(date);",
            "link": "https://stackoverflow.com/questions/50365049/unable-to-parse-date-using-datetimeformatter",
            "title": "Unable to parse date using DateTimeFormatter",
            "body": "<p>I am trying to parse a date string returned to me from DB to LocalDate using DateTimeFormatter. I am receiving the below exception.</p>\n\n<pre><code>String date = \"2018-05-16 03:39:13.0\";\n\nDateTimeFormatter formatter = DateTimeFormatter.ofPattern(\"yyyy-MM-dd HH:mm:ss.SSS\");\nLocalDateTime localDate= LocalDateTime.parse(date , formatter);\n</code></pre>\n\n<hr>\n\n<pre><code>java.time.format.DateTimeParseException: Text '2018-05-16 03:39:13.0' could not be parsed at index 20\n        at java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:1949) [rt.jar:1.8.0_171]\n        at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1851) [rt.jar:1.8.0_171]\n        at java.time.LocalDateTime.parse(LocalDateTime.java:492) [rt.jar:1.8.0_171]\n</code></pre>\n\n<p>However the below code is working.</p>\n\n<pre><code>String date = \"2018-05-16 03:39:13.0\";\nSimpleDateFormat formatter = new SimpleDateFormat(\"yyyy-MM-dd HH:mm:ss.SSS\");\nDate newDate= formatter.parse(date);\n</code></pre>\n"
        },
        {
            "tags": [
                "sql",
                "asp.net"
            ],
            "owner": {
                "reputation": 1,
                "user_id": 8746511,
                "user_type": "registered",
                "profile_image": "https://lh3.googleusercontent.com/-RTumioZFcy4/AAAAAAAAAAI/AAAAAAAAAdM/lIHImkmdpFI/photo.jpg?sz=128",
                "display_name": "rap per",
                "link": "https://stackoverflow.com/users/8746511/rap-per"
            },
            "is_answered": false,
            "view_count": 19,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1526463623,
            "creation_date": 1526463623,
            "question_id": 50367376,
            "body_markdown": "&gt; table 1 plan [enter image description here][1]\r\n&gt; table 2 logplan [enter image description here][2]\r\n  [1]: https://i.stack.imgur.com/mih9y.png\r\n  [2]: https://i.stack.imgur.com/U7ULg.png\r\n&gt; \r\n&gt; code query me \r\n\r\n    declare @Dateto datetime;\r\n    set @Dateto = &#39;03/03/2018&#39;;\r\n    \r\n     \r\n    select &#39;&#39; AS monthly\r\n    ,(select sum(pd.HeadCountPlan) as plang from ManpowerPlanDetail as pd\r\n    left outer join ManpowerPlanHead as ph on ph.ManpowerPlanHeadKey = pd.ManpowerPlanHeadKey\r\n    where ph.WorkPlaceCode = 2 and ph.PlanCode = 3 and ph.PartCode = 1\r\n    and convert(varchar(6),pd.UpdateTime, 112) &lt;= convert(varchar(6),@DateTo, 112)) AS planD\r\n    ,(select sum(pdl.HeadCountPlan) as plang \r\n    from ManpowerPlanDetailLog as pdl\r\n    left outer join ManpowerPlanDetail as pd on pd.ManpowerPlanDetailKey = pdl.ManpowerPlanDetailKey\r\n    left outer join ManpowerPlanHead as ph on ph.ManpowerPlanHeadKey = pd.ManpowerPlanHeadKey\r\n    where ph.WorkPlaceCode = 2 and ph.PlanCode = 3 and ph.PartCode = 1\r\n    and \r\n    (\r\n    \t(\r\n    \tCASE WHEN convert(varchar(6),@DateTo, 112) = convert(varchar(6),pdl.CreateTime, 112) THEN\r\n    \t convert(varchar(6),pdl.CreateTime, 112) END = convert(varchar(6),@DateTo, 112)\r\n    \t)\r\n    \tor\r\n    \t(\r\n    \tCASE WHEN convert(varchar(6),pdl.CreateTime, 112) != convert(varchar(6),@DateTo, 112) THEN \r\n    \t\tpdl.ManpowerPlanDetailLogKey END = (select top 1 ManpowerPlanDetailLogKey from ManpowerPlanDetailLog \r\n    \twhere ManpowerPlanDetailLog.ManpowerPlanDetailKey = pdl.ManpowerPlanDetailKey \r\n    \tand convert(varchar(6),ManpowerPlanDetailLog.CreateTime, 112) &lt; convert(varchar(6),@DateTo, 112)\r\n    \torder by ManpowerPlanDetailLogKey desc)\t\r\n    \t)\r\n    )\r\n    ) as planL\r\n\r\n&gt; I&#39;m in trouble Case when pdl.CreateTime = @Dateto and pdl.CreateTime != @datetime \r\n\r\n&gt; I want to do just one case when.\r\n&gt; \r\n&gt; I will bring  PlanD + PlanL = Plan  movement by month\r\n&gt; thank you",
            "link": "https://stackoverflow.com/questions/50367376/query-data-2-table-plan-log-plan-sum-data-by-month-where-id",
            "title": "query data 2 table Plan &amp; Log Plan sum data by month where id",
            "body": "<blockquote>\n  <p>table 1 plan <a href=\"https://i.stack.imgur.com/mih9y.png\" rel=\"nofollow noreferrer\">enter image description here</a>\n  table 2 logplan <a href=\"https://i.stack.imgur.com/U7ULg.png\" rel=\"nofollow noreferrer\">enter image description here</a></p>\n  \n  <p>code query me </p>\n</blockquote>\n\n<pre><code>declare @Dateto datetime;\nset @Dateto = '03/03/2018';\n\n\nselect '' AS monthly\n,(select sum(pd.HeadCountPlan) as plang from ManpowerPlanDetail as pd\nleft outer join ManpowerPlanHead as ph on ph.ManpowerPlanHeadKey = pd.ManpowerPlanHeadKey\nwhere ph.WorkPlaceCode = 2 and ph.PlanCode = 3 and ph.PartCode = 1\nand convert(varchar(6),pd.UpdateTime, 112) &lt;= convert(varchar(6),@DateTo, 112)) AS planD\n,(select sum(pdl.HeadCountPlan) as plang \nfrom ManpowerPlanDetailLog as pdl\nleft outer join ManpowerPlanDetail as pd on pd.ManpowerPlanDetailKey = pdl.ManpowerPlanDetailKey\nleft outer join ManpowerPlanHead as ph on ph.ManpowerPlanHeadKey = pd.ManpowerPlanHeadKey\nwhere ph.WorkPlaceCode = 2 and ph.PlanCode = 3 and ph.PartCode = 1\nand \n(\n    (\n    CASE WHEN convert(varchar(6),@DateTo, 112) = convert(varchar(6),pdl.CreateTime, 112) THEN\n     convert(varchar(6),pdl.CreateTime, 112) END = convert(varchar(6),@DateTo, 112)\n    )\n    or\n    (\n    CASE WHEN convert(varchar(6),pdl.CreateTime, 112) != convert(varchar(6),@DateTo, 112) THEN \n        pdl.ManpowerPlanDetailLogKey END = (select top 1 ManpowerPlanDetailLogKey from ManpowerPlanDetailLog \n    where ManpowerPlanDetailLog.ManpowerPlanDetailKey = pdl.ManpowerPlanDetailKey \n    and convert(varchar(6),ManpowerPlanDetailLog.CreateTime, 112) &lt; convert(varchar(6),@DateTo, 112)\n    order by ManpowerPlanDetailLogKey desc) \n    )\n)\n) as planL\n</code></pre>\n\n<blockquote>\n  <p>I'm in trouble Case when pdl.CreateTime = @Dateto and pdl.CreateTime != @datetime </p>\n  \n  <p>I want to do just one case when.</p>\n  \n  <p>I will bring  PlanD + PlanL = Plan  movement by month\n  thank you</p>\n</blockquote>\n"
        },
        {
            "tags": [
                "ios",
                "objective-c",
                "xamarin",
                "xamarin.forms",
                "xamarin.ios"
            ],
            "owner": {
                "reputation": 6,
                "user_id": 9535029,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/bf0c25357e3516067e021ca57ae56012?s=128&d=identicon&r=PG&f=1",
                "display_name": "Vishnu",
                "link": "https://stackoverflow.com/users/9535029/vishnu"
            },
            "is_answered": false,
            "view_count": 49,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1526463621,
            "creation_date": 1526381901,
            "last_edit_date": 1526463621,
            "question_id": 50348482,
            "body_markdown": "In my Xamarin.iOS project I used SecRecord/SecKeyChain to store my token values and app version. From production log I found keychain related exceptions with status code &#39;InteractionNotAllowed&#39;  when try to write/read items in keychain. Apple documents states that to resolve InteractionNotAllowed error we need to change the default kSecAttrAccessible attribute value from ‘**WhenUnlocked**&#39; to ‘**Always**’. \r\nBut in my existing code when I changed accessible attribute to &#39;**Always**&#39; app log out because it failed to read token from keychain. It return’s &#39;**Item not found**&#39; when read. But when I tried to save token again it returns &#39;**Duplicate item**&#39;. So again I tried to remove same item but this time it again returns &#39;**Item not found**&#39;. That’s really strange I can’t delete it and I can’t read it with same key. \r\n\r\nBelow is the code snippet - \r\n\r\n\r\n\r\n\r\n    private SecRecord CreateRecordForNewKeyValue(string accountName, string value)\r\n            {\r\n                return new SecRecord(SecKind.GenericPassword)\r\n                {\r\n                    Service = App.AppName,\r\n                    Account = accountName,\r\n                    ValueData = NSData.FromString(value, NSStringEncoding.UTF8),\r\n                    Accessible = SecAccessible.Always //This line of code is newly added.\r\n                };\r\n            }\r\n    \r\n    \r\n    \r\n    private SecRecord ExistingRecordForKey(string accountName)\r\n            {\r\n                return new SecRecord(SecKind.GenericPassword)\r\n                {\r\n                    Service = App.AppName,\r\n                    Account = accountName,\r\n                    Accessible = SecAccessible.Always //This line of code is newly added. \r\n                };\r\n            }\r\n    \r\n \r\n\r\n    public void SetValueForKeyAndAccount(string value, string accountName, string key)\r\n            {\r\n                var record = ExistingRecordForKey(accountName);\r\n                try\r\n                {\r\n                    if (string.IsNullOrEmpty(value))\r\n                    {\r\n                        if (!string.IsNullOrEmpty(GetValueFromAccountAndKey(accountName, key)))\r\n                            RemoveRecord(record);\r\n                        return;\r\n                    }\r\n                    // if the key already exists, remove it before set value\r\n                    if (!string.IsNullOrEmpty(GetValueFromAccountAndKey(accountName, key)))\r\n                        RemoveRecord(record);\r\n                }\r\n                catch (Exception e)\r\n                {\r\n                    //Log exception here -(&quot;RemoveRecord Failed &quot; + accountName, e,);\r\n                }\r\n                //Adding new record values to keychain\r\n                var result = SecKeyChain.Add(CreateRecordForNewKeyValue(accountName, value));\r\n                if (result != SecStatusCode.Success)\r\n                {\r\n                    if (result == SecStatusCode.DuplicateItem)\r\n                    {\r\n                        try\r\n                        {\r\n                            //Log exception here -(&quot;Error adding record: {0} for Account-&quot; + accountName, result), &quot;Try Remove account&quot;);\r\n                            RemoveRecord(record);\r\n                        }\r\n                        catch (Exception e)\r\n                        {\r\n                            //Log exception here -(&quot;RemoveRecord Failed  after getting error SecStatusCode.DuplicateItem for Account-&quot; + accountName, e);\r\n                        } \r\n                    }\r\n                    else\r\n                        throw new Exception(string.Format(&quot;Error adding record: {0} for Account-&quot; + accountName, result));\r\n                }\r\n            }    \r\n        \r\n\r\n    public string GetValueFromAccountAndKey(string accountName, string key)\r\n            {\r\n                try\r\n                {\r\n                    var record = ExistingRecordForKey(accountName);\r\n                    SecStatusCode resultCode;\r\n                    var match = SecKeyChain.QueryAsRecord(record, out resultCode);\r\n                    if (resultCode == SecStatusCode.Success)\r\n                    {\r\n                        if (match.ValueData != null)\r\n                        {\r\n                            string valueData = NSString.FromData(match.ValueData, NSStringEncoding.UTF8);\r\n                            if (string.IsNullOrEmpty(valueData))\r\n                                return string.Empty;\r\n                            return valueData;\r\n                        }\r\n                        else if (match.Generic != null)\r\n                        {\r\n                            string valueData = NSString.FromData(match.ValueData, NSStringEncoding.UTF8);\r\n                            if (string.IsNullOrEmpty(valueData))\r\n                                return string.Empty;\r\n                            return valueData;\r\n                        }\r\n                        else\r\n                            return string.Empty;\r\n                    }\r\n                }\r\n                catch (Exception e)\r\n                {\r\n                    // Exception logged here -(&quot;iOS Keychain Error for account-&quot; + accountName, e);\r\n                }\r\n                return string.Empty;\r\n            }\r\n\r\n   \r\n\r\nAny help would be great! Thanks \r\n\r\n",
            "link": "https://stackoverflow.com/questions/50348482/how-to-resolve-xamarin-ios-seckeychain-interactionnotallowed-issue",
            "title": "How to resolve Xamarin iOS SecKeyChain InteractionNotAllowed issue?",
            "body": "<p>In my Xamarin.iOS project I used SecRecord/SecKeyChain to store my token values and app version. From production log I found keychain related exceptions with status code 'InteractionNotAllowed'  when try to write/read items in keychain. Apple documents states that to resolve InteractionNotAllowed error we need to change the default kSecAttrAccessible attribute value from ‘<strong>WhenUnlocked</strong>' to ‘<strong>Always</strong>’. \nBut in my existing code when I changed accessible attribute to '<strong>Always</strong>' app log out because it failed to read token from keychain. It return’s '<strong>Item not found</strong>' when read. But when I tried to save token again it returns '<strong>Duplicate item</strong>'. So again I tried to remove same item but this time it again returns '<strong>Item not found</strong>'. That’s really strange I can’t delete it and I can’t read it with same key. </p>\n\n<p>Below is the code snippet - </p>\n\n<pre><code>private SecRecord CreateRecordForNewKeyValue(string accountName, string value)\n        {\n            return new SecRecord(SecKind.GenericPassword)\n            {\n                Service = App.AppName,\n                Account = accountName,\n                ValueData = NSData.FromString(value, NSStringEncoding.UTF8),\n                Accessible = SecAccessible.Always //This line of code is newly added.\n            };\n        }\n\n\n\nprivate SecRecord ExistingRecordForKey(string accountName)\n        {\n            return new SecRecord(SecKind.GenericPassword)\n            {\n                Service = App.AppName,\n                Account = accountName,\n                Accessible = SecAccessible.Always //This line of code is newly added. \n            };\n        }\n\n\n\npublic void SetValueForKeyAndAccount(string value, string accountName, string key)\n        {\n            var record = ExistingRecordForKey(accountName);\n            try\n            {\n                if (string.IsNullOrEmpty(value))\n                {\n                    if (!string.IsNullOrEmpty(GetValueFromAccountAndKey(accountName, key)))\n                        RemoveRecord(record);\n                    return;\n                }\n                // if the key already exists, remove it before set value\n                if (!string.IsNullOrEmpty(GetValueFromAccountAndKey(accountName, key)))\n                    RemoveRecord(record);\n            }\n            catch (Exception e)\n            {\n                //Log exception here -(\"RemoveRecord Failed \" + accountName, e,);\n            }\n            //Adding new record values to keychain\n            var result = SecKeyChain.Add(CreateRecordForNewKeyValue(accountName, value));\n            if (result != SecStatusCode.Success)\n            {\n                if (result == SecStatusCode.DuplicateItem)\n                {\n                    try\n                    {\n                        //Log exception here -(\"Error adding record: {0} for Account-\" + accountName, result), \"Try Remove account\");\n                        RemoveRecord(record);\n                    }\n                    catch (Exception e)\n                    {\n                        //Log exception here -(\"RemoveRecord Failed  after getting error SecStatusCode.DuplicateItem for Account-\" + accountName, e);\n                    } \n                }\n                else\n                    throw new Exception(string.Format(\"Error adding record: {0} for Account-\" + accountName, result));\n            }\n        }    \n\n\npublic string GetValueFromAccountAndKey(string accountName, string key)\n        {\n            try\n            {\n                var record = ExistingRecordForKey(accountName);\n                SecStatusCode resultCode;\n                var match = SecKeyChain.QueryAsRecord(record, out resultCode);\n                if (resultCode == SecStatusCode.Success)\n                {\n                    if (match.ValueData != null)\n                    {\n                        string valueData = NSString.FromData(match.ValueData, NSStringEncoding.UTF8);\n                        if (string.IsNullOrEmpty(valueData))\n                            return string.Empty;\n                        return valueData;\n                    }\n                    else if (match.Generic != null)\n                    {\n                        string valueData = NSString.FromData(match.ValueData, NSStringEncoding.UTF8);\n                        if (string.IsNullOrEmpty(valueData))\n                            return string.Empty;\n                        return valueData;\n                    }\n                    else\n                        return string.Empty;\n                }\n            }\n            catch (Exception e)\n            {\n                // Exception logged here -(\"iOS Keychain Error for account-\" + accountName, e);\n            }\n            return string.Empty;\n        }\n</code></pre>\n\n<p>Any help would be great! Thanks </p>\n"
        },
        {
            "tags": [
                "objective-c",
                "xcode",
                "utf-8",
                "swift4",
                "nsdata"
            ],
            "owner": {
                "reputation": 3,
                "user_id": 8064467,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/bf1b044c276a910093316c88e1f0f481?s=128&d=identicon&r=PG&f=1",
                "display_name": "sagar ss",
                "link": "https://stackoverflow.com/users/8064467/sagar-ss"
            },
            "is_answered": false,
            "view_count": 29,
            "answer_count": 1,
            "score": -2,
            "last_activity_date": 1526463616,
            "creation_date": 1526453036,
            "last_edit_date": 1526463616,
            "question_id": 50364043,
            "body_markdown": "Can anyone please help me convert this objective-c function to swift4?\r\nI have tried Swiftify but it is giving wrong output. Also I have tried with string to data conversion but output is wrong.\r\n\r\n\r\n    - (NSData *)stringToData{\r\n    if (!HCKBluetoothValidStr(self)) {\r\n        return nil;\r\n    }\r\n    if (!(self.length % 2 == 0)) {\r\n        return nil;\r\n    }\r\n    Byte bytes[255] = {0};\r\n    NSInteger count = 0;\r\n    for (int i =0; i &lt; self.length; i+=2) {\r\n        \r\n        NSString *strByte = [self substringWithRange:NSMakeRange(i,2)];\r\n        NSLog(@&quot;strbyte is: %@ \\n\\n&quot;,strByte);\r\n        unsigned long red = strtoul([strByte UTF8String],0,16);\r\n        Byte b =  (Byte) ((0xff &amp; red) );//( Byte) 0xff&amp;iByte;\r\n        bytes[i/2+0] = b;\r\n        count ++;\r\n    }\r\n    NSData * data = [NSData dataWithBytes:bytes length:count];\r\n    return data;\r\n}\r\n\r\nfor same function input string is:\r\n\r\n    NSString *commandStr = @&quot;1702030a0a&quot;;\r\n\r\nand output from function in NSData format is:\r\n\r\n    &lt;1702030a 0a&gt;\r\n\r\n# EDIT 1: #\r\nI have tried to convert it to swift, but strString.utf8CString giving error as &quot;Cannot convert value of type &#39;ContiguousArray&lt;CChar&gt;&#39; (aka &#39;ContiguousArray&lt;Int8&gt;&#39;) to expected argument type &#39;UnsafePointer&lt;Int8&gt;!&#39;&quot;\r\n\r\n\r\n    func stringToData(stringVal: NSString) -&gt; NSData {\r\n        var bytes: [UInt8] = [0]\r\n        var count: Int = 0\r\n        var i = 0\r\n        for index in stride(from: 0, to: stringVal.length , by: 2) {\r\n            var strByte: NSString = stringVal.substring(with: NSMakeRange(index, 2)) as NSString\r\n            print(&quot;strbyte is: \\(strByte) \\n\\n&quot;)\r\n            let strString = String(strByte)\r\n            print(strString)\r\n            var red: UInt8 = strtoul(strString.utf8CString, 0, 16)\r\n            \r\n            \r\n            let b = UInt8((0xff &amp; red))\r\n            count = count+1\r\n        }\r\n        var data = NSData(bytes: bytes, length: count)\r\n        return data\r\n    }",
            "link": "https://stackoverflow.com/questions/50364043/not-able-to-convert-objective-c-code-to-swift-4",
            "title": "Not able to convert objective-c code to swift 4",
            "body": "<p>Can anyone please help me convert this objective-c function to swift4?\nI have tried Swiftify but it is giving wrong output. Also I have tried with string to data conversion but output is wrong.</p>\n\n<pre><code>- (NSData *)stringToData{\nif (!HCKBluetoothValidStr(self)) {\n    return nil;\n}\nif (!(self.length % 2 == 0)) {\n    return nil;\n}\nByte bytes[255] = {0};\nNSInteger count = 0;\nfor (int i =0; i &lt; self.length; i+=2) {\n\n    NSString *strByte = [self substringWithRange:NSMakeRange(i,2)];\n    NSLog(@\"strbyte is: %@ \\n\\n\",strByte);\n    unsigned long red = strtoul([strByte UTF8String],0,16);\n    Byte b =  (Byte) ((0xff &amp; red) );//( Byte) 0xff&amp;iByte;\n    bytes[i/2+0] = b;\n    count ++;\n}\nNSData * data = [NSData dataWithBytes:bytes length:count];\nreturn data;\n</code></pre>\n\n<p>}</p>\n\n<p>for same function input string is:</p>\n\n<pre><code>NSString *commandStr = @\"1702030a0a\";\n</code></pre>\n\n<p>and output from function in NSData format is:</p>\n\n<pre><code>&lt;1702030a 0a&gt;\n</code></pre>\n\n<h1>EDIT 1:</h1>\n\n<p>I have tried to convert it to swift, but strString.utf8CString giving error as \"Cannot convert value of type 'ContiguousArray' (aka 'ContiguousArray') to expected argument type 'UnsafePointer!'\"</p>\n\n<pre><code>func stringToData(stringVal: NSString) -&gt; NSData {\n    var bytes: [UInt8] = [0]\n    var count: Int = 0\n    var i = 0\n    for index in stride(from: 0, to: stringVal.length , by: 2) {\n        var strByte: NSString = stringVal.substring(with: NSMakeRange(index, 2)) as NSString\n        print(\"strbyte is: \\(strByte) \\n\\n\")\n        let strString = String(strByte)\n        print(strString)\n        var red: UInt8 = strtoul(strString.utf8CString, 0, 16)\n\n\n        let b = UInt8((0xff &amp; red))\n        count = count+1\n    }\n    var data = NSData(bytes: bytes, length: count)\n    return data\n}\n</code></pre>\n"
        },
        {
            "tags": [
                ".htaccess",
                "redirect",
                "hosting"
            ],
            "owner": {
                "reputation": 5,
                "user_id": 4793784,
                "user_type": "registered",
                "accept_rate": 25,
                "profile_image": "https://www.gravatar.com/avatar/4970bcc9ad490bc7d07c9635889b308e?s=128&d=identicon&r=PG",
                "display_name": "Michael Kopytin",
                "link": "https://stackoverflow.com/users/4793784/michael-kopytin"
            },
            "is_answered": false,
            "view_count": 9,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1526463610,
            "creation_date": 1526463610,
            "question_id": 50367372,
            "body_markdown": "I have a two language website. The one language is located in **/en/** folder, the other in **/ru/**. I need to make server to redirect to **/ru/** whenever i go to **/**. Last time I made htaccess file to do such thing the server responded with something like &quot;The website redirects the way that will never end redirecting&quot;. Basically I need the solution that would work.",
            "link": "https://stackoverflow.com/questions/50367372/redirect-from-index-html-to-en-index-html",
            "title": "Redirect from /index.html to /en/index.html",
            "body": "<p>I have a two language website. The one language is located in <strong>/en/</strong> folder, the other in <strong>/ru/</strong>. I need to make server to redirect to <strong>/ru/</strong> whenever i go to <strong>/</strong>. Last time I made htaccess file to do such thing the server responded with something like \"The website redirects the way that will never end redirecting\". Basically I need the solution that would work.</p>\n"
        },
        {
            "tags": [
                "java",
                "structure",
                "naming-conventions",
                "packages"
            ],
            "owner": {
                "reputation": 399,
                "user_id": 1432310,
                "user_type": "registered",
                "accept_rate": 42,
                "profile_image": "https://www.gravatar.com/avatar/46d1ee2a8b1c628383e40ff901897d59?s=128&d=identicon&r=PG",
                "display_name": "ABR",
                "link": "https://stackoverflow.com/users/1432310/abr"
            },
            "is_answered": true,
            "view_count": 23,
            "accepted_answer_id": 50367366,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1526463594,
            "creation_date": 1526463335,
            "question_id": 50367261,
            "body_markdown": "Is it considered a bad practice in java to have a structure where some package/s contain both classes and sub packages? i.e :\r\n\r\n    com/\r\n      domain/\r\n         SomeClass.java\r\n         AnotherClass.java\r\n         model/\r\n           Model1.java\r\n           Model2.java\r\n         somePackage/\r\n            SomePackageClass.java\r\n\r\n\r\nThe other option being `SomeClass.java` and `AnotherClass.java` contained in third package\r\n\r\n",
            "link": "https://stackoverflow.com/questions/50367261/is-it-considered-bad-practice-in-java-for-a-package-to-contain-both-classes-and",
            "title": "Is it considered bad practice in java for a package to contain both classes and sub-packages?",
            "body": "<p>Is it considered a bad practice in java to have a structure where some package/s contain both classes and sub packages? i.e :</p>\n\n<pre><code>com/\n  domain/\n     SomeClass.java\n     AnotherClass.java\n     model/\n       Model1.java\n       Model2.java\n     somePackage/\n        SomePackageClass.java\n</code></pre>\n\n<p>The other option being <code>SomeClass.java</code> and <code>AnotherClass.java</code> contained in third package</p>\n"
        },
        {
            "tags": [
                "java",
                "sparql",
                "rdf",
                "jena"
            ],
            "owner": {
                "reputation": 67,
                "user_id": 9443932,
                "user_type": "registered",
                "accept_rate": 75,
                "profile_image": "https://www.gravatar.com/avatar/3056b8712e86cede3288d8ffcbdee239?s=128&d=identicon&r=PG&f=1",
                "display_name": "vimalDev",
                "link": "https://stackoverflow.com/users/9443932/vimaldev"
            },
            "is_answered": true,
            "view_count": 64,
            "accepted_answer_id": 50367363,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1526463583,
            "creation_date": 1522060618,
            "question_id": 49489422,
            "body_markdown": "I am getting `Exception in thread &quot;main&quot; com.hp.hpl.jena.query.QueryParseException: Lexical error at line 1, column 176.  Encountered: &quot; &quot; (32), after: &quot;BIND&quot;` error while running below query using Jena,\r\n\r\n    String queryString = &quot;PREFIX  ns:  &lt;http://example.org/ns#&gt;&quot; + \r\n    \t\t\t\t&quot;SELECT  ?title ?price&quot; + \r\n    \t\t\t\t&quot;  {&quot; + \r\n    \t\t\t\t&quot;    ?x ns:price ?p .&quot; + \r\n    \t\t\t\t&quot;    ?x ns:discount ?discount .&quot; + \r\n    \t\t\t\t&quot;    ?x &lt;http://purl.org/dc/elements/1.1/title&gt; ?title . &quot; + \r\n    \t\t\t\t&quot;    BIND (&#39;12&#39;^^xsd:integer AS ?price)&quot; + \r\n    \t\t\t\t&quot;    FILTER( ?price &lt; 20 )&quot; + \r\n    \t\t\t\t&quot;  }&quot;;\r\n\r\nBelow is my stack trace,\r\n\r\n    Exception in thread &quot;main&quot; com.hp.hpl.jena.query.QueryParseException: Lexical error at line 1, column 176.  Encountered: &quot; &quot; (32), after : &quot;BIND&quot;\r\n    \tat com.hp.hpl.jena.sparql.lang.ParserSPARQL10.perform(ParserSPARQL10.java:95)\r\n    \tat com.hp.hpl.jena.sparql.lang.ParserSPARQL10.parse(ParserSPARQL10.java:39)\r\n    \tat com.hp.hpl.jena.query.QueryFactory.parse(QueryFactory.java:129)\r\n    \tat com.hp.hpl.jena.query.QueryFactory.create(QueryFactory.java:72)\r\n    \tat com.hp.hpl.jena.query.QueryFactory.create(QueryFactory.java:43)\r\n    \tat com.hp.hpl.jena.query.QueryFactory.create(QueryFactory.java:31)\r\n    \tat com.hp.hpl.jena.query.QueryExecutionFactory.sparqlService(QueryExecutionFactory.java:342)\r\n    \tat com.ifour.testdata.AW317.QuerySampleDataSPARQL.main(QuerySampleDataSPARQL.java:28)\r\n\r\ncan anybody please help me on an issue I have googled but didn&#39;t find the solution.The same query runs on virtuoso SPARQL endpoint giving me perfect result.but while using java Jena SPARQL query I am getting this error.\r\n",
            "link": "https://stackoverflow.com/questions/49489422/exception-in-thread-main-com-hp-hpl-jena-query-queryparseexception-lexical-er",
            "title": "Exception in thread &quot;main&quot; com.hp.hpl.jena.query.QueryParseException: Lexical error at line 1, column 176. Encountered: &quot; &quot; (32), after : &quot;BIND&quot;",
            "body": "<p>I am getting <code>Exception in thread \"main\" com.hp.hpl.jena.query.QueryParseException: Lexical error at line 1, column 176.  Encountered: \" \" (32), after: \"BIND\"</code> error while running below query using Jena,</p>\n\n<pre><code>String queryString = \"PREFIX  ns:  &lt;http://example.org/ns#&gt;\" + \n                \"SELECT  ?title ?price\" + \n                \"  {\" + \n                \"    ?x ns:price ?p .\" + \n                \"    ?x ns:discount ?discount .\" + \n                \"    ?x &lt;http://purl.org/dc/elements/1.1/title&gt; ?title . \" + \n                \"    BIND ('12'^^xsd:integer AS ?price)\" + \n                \"    FILTER( ?price &lt; 20 )\" + \n                \"  }\";\n</code></pre>\n\n<p>Below is my stack trace,</p>\n\n<pre><code>Exception in thread \"main\" com.hp.hpl.jena.query.QueryParseException: Lexical error at line 1, column 176.  Encountered: \" \" (32), after : \"BIND\"\n    at com.hp.hpl.jena.sparql.lang.ParserSPARQL10.perform(ParserSPARQL10.java:95)\n    at com.hp.hpl.jena.sparql.lang.ParserSPARQL10.parse(ParserSPARQL10.java:39)\n    at com.hp.hpl.jena.query.QueryFactory.parse(QueryFactory.java:129)\n    at com.hp.hpl.jena.query.QueryFactory.create(QueryFactory.java:72)\n    at com.hp.hpl.jena.query.QueryFactory.create(QueryFactory.java:43)\n    at com.hp.hpl.jena.query.QueryFactory.create(QueryFactory.java:31)\n    at com.hp.hpl.jena.query.QueryExecutionFactory.sparqlService(QueryExecutionFactory.java:342)\n    at com.ifour.testdata.AW317.QuerySampleDataSPARQL.main(QuerySampleDataSPARQL.java:28)\n</code></pre>\n\n<p>can anybody please help me on an issue I have googled but didn't find the solution.The same query runs on virtuoso SPARQL endpoint giving me perfect result.but while using java Jena SPARQL query I am getting this error.</p>\n"
        },
        {
            "tags": [
                "validation",
                "laravel-5",
                "ip-address"
            ],
            "owner": {
                "reputation": 159,
                "user_id": 1257371,
                "user_type": "registered",
                "accept_rate": 37,
                "profile_image": "https://www.gravatar.com/avatar/98a58d19e44b73d2e49809555181ad96?s=128&d=identicon&r=PG",
                "display_name": "Rizwan Ranjha",
                "link": "https://stackoverflow.com/users/1257371/rizwan-ranjha"
            },
            "is_answered": true,
            "view_count": 1563,
            "answer_count": 2,
            "score": 1,
            "last_activity_date": 1526463577,
            "creation_date": 1442044613,
            "question_id": 32536549,
            "body_markdown": "I am trying to find a method to validate IP Address before submitting this to my database, I have tried a lot of Google Search but unable to do this in my Laravel Controller. Can you please help? Appreciated",
            "link": "https://stackoverflow.com/questions/32536549/ip-address-validation-in-controller-in-laravel-5",
            "title": "IP Address Validation in Controller in Laravel 5",
            "body": "<p>I am trying to find a method to validate IP Address before submitting this to my database, I have tried a lot of Google Search but unable to do this in my Laravel Controller. Can you please help? Appreciated</p>\n"
        },
        {
            "tags": [
                "java",
                "spring-boot",
                "jvm"
            ],
            "owner": {
                "reputation": 1566,
                "user_id": 355634,
                "user_type": "registered",
                "accept_rate": 50,
                "profile_image": "https://www.gravatar.com/avatar/ea46acb20e91f4143e84ed750c66efb2?s=128&d=identicon&r=PG",
                "display_name": "Udara S.S Liyanage",
                "link": "https://stackoverflow.com/users/355634/udara-s-s-liyanage"
            },
            "is_answered": true,
            "view_count": 337,
            "answer_count": 2,
            "score": 2,
            "last_activity_date": 1526463574,
            "creation_date": 1492779657,
            "last_edit_date": 1526463574,
            "question_id": 43543404,
            "body_markdown": "I have a Springboot application with Mule running inside a docker container as a micro service. It takes about 700MB even when it is idle. It is noticed that JVM has allocated heap of 380 MB which is max heap provided using `-Xmx` parameter. Though max heap is allocated, the micro service only use about 50 MB when idle. The question is how to released unused memory back from JVM.\r\n\r\nIt seems that reducing MaxHeapFreeRatio we can ask JVM to shrink when there is more free memory ratio. However `MinHeapFreeRatio=20 -XX:MaxHeapFreeRatio=40` did not make much difference and JVM did not release memory. But when I use `-Xmn` with above two parameters JVM releases heap memory as expected. See below image for example scenario.\r\n\r\n#Java version 8\r\n\r\n    -Xmn100M  -XX:+PrintGCDetails  -XX:MinHeapFreeRatio=20 -XX:MaxHeapFreeRatio=40 -Xmx384M\r\n\r\n[![enter image description here][1]][1]\r\n\r\n 1. Why does MinHeapFreeRatio and MaxHeapFreeRatio does not work as expected?\r\n 2. If above parameters are fine what are the consequences of `-Xmn` and what is the value of it should be?\r\n 3. What are other solutions in order to achieve the task of releasing memory back from JVM?\r\n\r\n  [1]: https://i.stack.imgur.com/Jkf5l.png",
            "link": "https://stackoverflow.com/questions/43543404/jvm-release-unused-heap",
            "title": "JVM release unused heap",
            "body": "<p>I have a Springboot application with Mule running inside a docker container as a micro service. It takes about 700MB even when it is idle. It is noticed that JVM has allocated heap of 380 MB which is max heap provided using <code>-Xmx</code> parameter. Though max heap is allocated, the micro service only use about 50 MB when idle. The question is how to released unused memory back from JVM.</p>\n\n<p>It seems that reducing MaxHeapFreeRatio we can ask JVM to shrink when there is more free memory ratio. However <code>MinHeapFreeRatio=20 -XX:MaxHeapFreeRatio=40</code> did not make much difference and JVM did not release memory. But when I use <code>-Xmn</code> with above two parameters JVM releases heap memory as expected. See below image for example scenario.</p>\n\n<h1>Java version 8</h1>\n\n<pre><code>-Xmn100M  -XX:+PrintGCDetails  -XX:MinHeapFreeRatio=20 -XX:MaxHeapFreeRatio=40 -Xmx384M\n</code></pre>\n\n<p><a href=\"https://i.stack.imgur.com/Jkf5l.png\" rel=\"nofollow noreferrer\"><img src=\"https://i.stack.imgur.com/Jkf5l.png\" alt=\"enter image description here\"></a></p>\n\n<ol>\n<li>Why does MinHeapFreeRatio and MaxHeapFreeRatio does not work as expected?</li>\n<li>If above parameters are fine what are the consequences of <code>-Xmn</code> and what is the value of it should be?</li>\n<li>What are other solutions in order to achieve the task of releasing memory back from JVM?</li>\n</ol>\n"
        },
        {
            "tags": [
                "django",
                "python-3.x"
            ],
            "owner": {
                "reputation": 1,
                "user_id": 8603544,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/4af0e8f3ff6e3fad90fa75c99c368418?s=128&d=identicon&r=PG&f=1",
                "display_name": "M511",
                "link": "https://stackoverflow.com/users/8603544/m511"
            },
            "is_answered": false,
            "view_count": 24,
            "answer_count": 0,
            "score": -2,
            "last_activity_date": 1526463570,
            "creation_date": 1526459906,
            "last_edit_date": 1526463570,
            "question_id": 50366079,
            "body_markdown": "is there a way to install Django even behind a corporate firewall?\r\n\r\nI keep getting the following error using pip install:\r\n\r\n&gt; Could not fetch URL https://pypi.python.org/simple/django/: There as a\r\n&gt; problem confirming the ssl cerificate: [SSL:\r\n&gt; CERTIFICATE_VERIFY_FAILED] certificate very failed (_ssl.c:465) -\r\n&gt; skipping\r\n&gt; \r\n&gt; Could not find a version that satisfies the requirement Django==2.0.5\r\n&gt; (from versions: )\r\n&gt; \r\n&gt; No matching distributions found for Django==2.0.5\r\n\r\nI was able to install this version on my personal computer easily. The machine with the problem is using Pythong 3.52.:: Anaconda 4.2.0 (and no I can&#39;t upgrade both of those).\r\n",
            "link": "https://stackoverflow.com/questions/50366079/installing-django-behind-corporate-firewall",
            "title": "Installing Django behind corporate firewall",
            "body": "<p>is there a way to install Django even behind a corporate firewall?</p>\n\n<p>I keep getting the following error using pip install:</p>\n\n<blockquote>\n  <p>Could not fetch URL <a href=\"https://pypi.python.org/simple/django/\" rel=\"nofollow noreferrer\">https://pypi.python.org/simple/django/</a>: There as a\n  problem confirming the ssl cerificate: [SSL:\n  CERTIFICATE_VERIFY_FAILED] certificate very failed (_ssl.c:465) -\n  skipping</p>\n  \n  <p>Could not find a version that satisfies the requirement Django==2.0.5\n  (from versions: )</p>\n  \n  <p>No matching distributions found for Django==2.0.5</p>\n</blockquote>\n\n<p>I was able to install this version on my personal computer easily. The machine with the problem is using Pythong 3.52.:: Anaconda 4.2.0 (and no I can't upgrade both of those).</p>\n"
        },
        {
            "tags": [
                "google-cloud-functions",
                "google-cloud-dataflow"
            ],
            "owner": {
                "reputation": 22,
                "user_id": 7408930,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/b0edf0bde9d948112cdd4e1cae2944dc?s=128&d=identicon&r=PG&f=1",
                "display_name": "BenAhm",
                "link": "https://stackoverflow.com/users/7408930/benahm"
            },
            "is_answered": true,
            "view_count": 84,
            "answer_count": 2,
            "score": 0,
            "last_activity_date": 1526463564,
            "creation_date": 1525718281,
            "last_edit_date": 1526463564,
            "question_id": 50220510,
            "body_markdown": "I have a small pipeline im trying to execute:\r\n\r\n1. file placed into GCS Bucket &gt; 2. Cloud Function triggers Dataflow job when file is placed in GCS bucket (not working) &gt; 3. Writes to Big Query table (this part working)\r\n\r\nI&#39;ve created a Dataflow job through Dataprep as it has nice UI to do all my transformations before writing to a BigQuery table (writing to BigQuery works fine), and the Cloud function triggers when a file is uploaded to the GCS bucket. However the Cloud Function doesn&#39;t trigger the Dataflow job (which I wrote in Dataprep). \r\n\r\nPlease, have a look at my sample code below of my Cloud Function, if I can get any pointers as to why the Dataflow job is not triggering. \r\n\r\n&lt;!-- begin snippet: js hide: false console: true babel: false --&gt;\r\n\r\n&lt;!-- language: lang-html --&gt;\r\n\r\n    /**\r\n     * Triggered from a message on a Cloud Storage bucket.\r\n     *\r\n     * @param {!Object} event The Cloud Functions event.\r\n     * @param {!Function} The callback function.\r\n     */\r\n    exports.processFile = (event, callback) =&gt; {\r\n      console.log(&#39;Processing file: &#39; + event.data.name);\r\n      callback();\r\n\r\n      const google = require(&#39;googleapis&#39;);\r\n\r\n     exports.CF_GCStoDataFlow_v2 = function(event, callback) {\r\n      const file = event.data;\r\n      if (file.resourceState === &#39;exists&#39; &amp;&amp; file.name) {\r\n        google.auth.getApplicationDefault(function (err, authClient, projectId) {\r\n          if (err) {\r\n            throw err;\r\n          }\r\n\r\n          if (authClient.createScopedRequired &amp;&amp; authClient.createScopedRequired()) {\r\n            authClient = authClient.createScoped([\r\n              &#39;https://www.googleapis.com/auth/cloud-platform&#39;,\r\n              &#39;https://www.googleapis.com/auth/userinfo.email&#39;\r\n            ]);\r\n          }\r\n\r\n          const dataflow = google.dataflow({ version: &#39;v1b3&#39;, auth: authClient });\r\n\r\n          dataflow.projects.templates.create({\r\n            projectId: projectId,\r\n            resource: {\r\n              parameters: {\r\n                inputFile: `gs://${file.bucket}/${file.name}`,\r\n                outputFile: `gs://${file.bucket}/${file.name}`\r\n              },\r\n              jobName: &#39;cloud-dataprep-csvtobq-v2-281345&#39;,\r\n              gcsPath: &#39;gs://mygcstest-pipeline-staging/temp/&#39;\r\n            }\r\n          }, function(err, response) {\r\n            if (err) {\r\n              console.error(&quot;problem running dataflow template, error was: &quot;, err);\r\n            }\r\n            console.log(&quot;Dataflow template response: &quot;, response);\r\n            callback();\r\n          });\r\n\r\n        });\r\n      }\r\n     };\r\n    };\r\n\r\n&lt;!-- end snippet --&gt;\r\n\r\n[![DataProc job][1]][1]\r\n\r\n\r\n \r\n\r\n\r\n  [1]: https://i.stack.imgur.com/zEheG.png",
            "link": "https://stackoverflow.com/questions/50220510/cloud-function-to-trigger-dataprep-dataflow-job",
            "title": "Cloud Function to Trigger DataPrep Dataflow Job",
            "body": "<p>I have a small pipeline im trying to execute:</p>\n\n<ol>\n<li>file placed into GCS Bucket > 2. Cloud Function triggers Dataflow job when file is placed in GCS bucket (not working) > 3. Writes to Big Query table (this part working)</li>\n</ol>\n\n<p>I've created a Dataflow job through Dataprep as it has nice UI to do all my transformations before writing to a BigQuery table (writing to BigQuery works fine), and the Cloud function triggers when a file is uploaded to the GCS bucket. However the Cloud Function doesn't trigger the Dataflow job (which I wrote in Dataprep). </p>\n\n<p>Please, have a look at my sample code below of my Cloud Function, if I can get any pointers as to why the Dataflow job is not triggering. </p>\n\n<p><div class=\"snippet\" data-lang=\"js\" data-hide=\"false\" data-console=\"true\" data-babel=\"false\">\r\n<div class=\"snippet-code\">\r\n<pre class=\"snippet-code-html lang-html prettyprint-override\"><code>/**\r\n * Triggered from a message on a Cloud Storage bucket.\r\n *\r\n * @param {!Object} event The Cloud Functions event.\r\n * @param {!Function} The callback function.\r\n */\r\nexports.processFile = (event, callback) =&gt; {\r\n  console.log('Processing file: ' + event.data.name);\r\n  callback();\r\n\r\n  const google = require('googleapis');\r\n\r\n exports.CF_GCStoDataFlow_v2 = function(event, callback) {\r\n  const file = event.data;\r\n  if (file.resourceState === 'exists' &amp;&amp; file.name) {\r\n    google.auth.getApplicationDefault(function (err, authClient, projectId) {\r\n      if (err) {\r\n        throw err;\r\n      }\r\n\r\n      if (authClient.createScopedRequired &amp;&amp; authClient.createScopedRequired()) {\r\n        authClient = authClient.createScoped([\r\n          'https://www.googleapis.com/auth/cloud-platform',\r\n          'https://www.googleapis.com/auth/userinfo.email'\r\n        ]);\r\n      }\r\n\r\n      const dataflow = google.dataflow({ version: 'v1b3', auth: authClient });\r\n\r\n      dataflow.projects.templates.create({\r\n        projectId: projectId,\r\n        resource: {\r\n          parameters: {\r\n            inputFile: `gs://${file.bucket}/${file.name}`,\r\n            outputFile: `gs://${file.bucket}/${file.name}`\r\n          },\r\n          jobName: 'cloud-dataprep-csvtobq-v2-281345',\r\n          gcsPath: 'gs://mygcstest-pipeline-staging/temp/'\r\n        }\r\n      }, function(err, response) {\r\n        if (err) {\r\n          console.error(\"problem running dataflow template, error was: \", err);\r\n        }\r\n        console.log(\"Dataflow template response: \", response);\r\n        callback();\r\n      });\r\n\r\n    });\r\n  }\r\n };\r\n};</code></pre>\r\n</div>\r\n</div>\r\n</p>\n\n<p><a href=\"https://i.stack.imgur.com/zEheG.png\" rel=\"nofollow noreferrer\"><img src=\"https://i.stack.imgur.com/zEheG.png\" alt=\"DataProc job\"></a></p>\n"
        },
        {
            "tags": [
                "javascript",
                "office-js",
                "office-js-helpers"
            ],
            "owner": {
                "reputation": 34,
                "user_id": 4896473,
                "user_type": "registered",
                "accept_rate": 0,
                "profile_image": "https://www.gravatar.com/avatar/4a8d7f19212dd3719159055bbd009233?s=128&d=identicon&r=PG&f=1",
                "display_name": "Arupam Sengupta",
                "link": "https://stackoverflow.com/users/4896473/arupam-sengupta"
            },
            "is_answered": false,
            "view_count": 23,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1526463552,
            "creation_date": 1526463125,
            "last_edit_date": 1526463552,
            "question_id": 50367177,
            "body_markdown": "I was using the following function from Office js library to display a Excel dialog - \r\n\r\n    Office.context.ui.displayDialogAsync(startAddress, options, callback)\r\n\r\nOn closing the dialog box programatically and trying to open it again , it is showing the following error - \r\n\r\n&gt; Error code - 12007 .A dialog box is already opened from the task pane.\r\n&gt; A task pane add-in can only have one dialog box open at a time.\r\n\r\nThe issue is persisting at `Microsoft Excel 16.13`. It works fine on all the previous versions of Excel.",
            "link": "https://stackoverflow.com/questions/50367177/office-js-dialog-issue-in-excel-16-13",
            "title": "Office js Dialog issue in Excel 16.13",
            "body": "<p>I was using the following function from Office js library to display a Excel dialog - </p>\n\n<pre><code>Office.context.ui.displayDialogAsync(startAddress, options, callback)\n</code></pre>\n\n<p>On closing the dialog box programatically and trying to open it again , it is showing the following error - </p>\n\n<blockquote>\n  <p>Error code - 12007 .A dialog box is already opened from the task pane.\n  A task pane add-in can only have one dialog box open at a time.</p>\n</blockquote>\n\n<p>The issue is persisting at <code>Microsoft Excel 16.13</code>. It works fine on all the previous versions of Excel.</p>\n"
        },
        {
            "tags": [
                "64bit",
                "ubuntu-14.04",
                "kvm",
                "android-studio-2.0",
                "ubuntu-18.04"
            ],
            "owner": {
                "reputation": 81,
                "user_id": 6351152,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/70b2a1fc3425d549e67ea16beeb50ff5?s=128&d=identicon&r=PG&f=1",
                "display_name": "florian.R",
                "link": "https://stackoverflow.com/users/6351152/florian-r"
            },
            "is_answered": true,
            "view_count": 8696,
            "answer_count": 1,
            "score": 16,
            "last_activity_date": 1526463550,
            "creation_date": 1463577410,
            "last_edit_date": 1525769286,
            "question_id": 37300811,
            "body_markdown": "Noob question on Android Studio. I have Android Studio running on a Ubuntu box and connected through a windows PC through rdp. After compiling the code when trying to run it on an emulator I get this error that \r\n\r\n&gt; /dev/kvm permission denied. \r\n\r\nI checked the permissions \r\n``sudo adduser `id -un` kvm\r\nAdding user &#39;&lt;username&gt;&#39; to group &#39;kvm&#39; ...``. \r\nand it returned that I am already in the kvm group. Is it because I am connecting through rdp? Right now I don&#39;t have access to the remote box for a week to test otherwise. Any help is appreciated.",
            "link": "https://stackoverflow.com/questions/37300811/android-studio-dev-kvm-device-permission-denied",
            "title": "Android Studio: /dev/kvm device permission denied",
            "body": "<p>Noob question on Android Studio. I have Android Studio running on a Ubuntu box and connected through a windows PC through rdp. After compiling the code when trying to run it on an emulator I get this error that </p>\n\n<blockquote>\n  <p>/dev/kvm permission denied. </p>\n</blockquote>\n\n<p>I checked the permissions \n<code>sudo adduser `id -un` kvm\nAdding user '&lt;username&gt;' to group 'kvm' ...</code>. \nand it returned that I am already in the kvm group. Is it because I am connecting through rdp? Right now I don't have access to the remote box for a week to test otherwise. Any help is appreciated.</p>\n"
        },
        {
            "tags": [
                "apache",
                "hadoop",
                "hdfs",
                "flume"
            ],
            "owner": {
                "reputation": 51,
                "user_id": 3052368,
                "user_type": "registered",
                "accept_rate": 67,
                "profile_image": "https://www.gravatar.com/avatar/0f018a947c8dc646fa13dd720cd8613f?s=128&d=identicon&r=PG&f=1",
                "display_name": "TheNobleSix",
                "link": "https://stackoverflow.com/users/3052368/thenoblesix"
            },
            "is_answered": false,
            "view_count": 14,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1526463548,
            "creation_date": 1526463548,
            "question_id": 50367349,
            "body_markdown": "I&#39;m trying to ingest a csv file and filter the tuples inside it so I can obtain a file for each identifier and write the data on HDFS. I have a Docker container with HDFS already configured.\r\n\r\nI wrote a config file following the documentation for the flume Agent, but when I start it nothing happens.\r\n\r\nSo I saw the log file and there are a NoClassDefFoundError for the HDFS Sink. I read that I need to define the HADOOP_HOME variabile but how can I do it being Hadoop inside a container?\r\n\r\nThank you very much for your time! ",
            "link": "https://stackoverflow.com/questions/50367349/flume-spooling-directory-to-hdfs-in-docker-container",
            "title": "Flume: Spooling Directory to HDFS in Docker container",
            "body": "<p>I'm trying to ingest a csv file and filter the tuples inside it so I can obtain a file for each identifier and write the data on HDFS. I have a Docker container with HDFS already configured.</p>\n\n<p>I wrote a config file following the documentation for the flume Agent, but when I start it nothing happens.</p>\n\n<p>So I saw the log file and there are a NoClassDefFoundError for the HDFS Sink. I read that I need to define the HADOOP_HOME variabile but how can I do it being Hadoop inside a container?</p>\n\n<p>Thank you very much for your time! </p>\n"
        },
        {
            "tags": [
                "git",
                "git-submodules",
                "multiple-repositories"
            ],
            "owner": {
                "reputation": 2537,
                "user_id": 176611,
                "user_type": "registered",
                "accept_rate": 62,
                "profile_image": "https://www.gravatar.com/avatar/d57fa520c489a3231ea76e9127f29fb6?s=128&d=identicon&r=PG",
                "display_name": "Brent",
                "link": "https://stackoverflow.com/users/176611/brent"
            },
            "is_answered": true,
            "view_count": 70651,
            "accepted_answer_id": 1811738,
            "answer_count": 4,
            "score": 228,
            "last_activity_date": 1526463545,
            "creation_date": 1259387867,
            "last_edit_date": 1406056223,
            "question_id": 1811730,
            "body_markdown": "I have a Git media repository where I&#39;m keeping all of my JavaScript and CSS master files and scripts that I&#39;ll use on various projects. \r\n\r\nIf I create a new project that&#39;s in its own Git repository, how do I use JavaScript files from my media repository in my new project in a way that makes it so I don&#39;t have to update both copies of the script when I make changes? ",
            "link": "https://stackoverflow.com/questions/1811730/how-do-i-work-with-a-git-repository-within-another-repository",
            "title": "How do I work with a git repository within another repository?",
            "body": "<p>I have a Git media repository where I'm keeping all of my JavaScript and CSS master files and scripts that I'll use on various projects. </p>\n\n<p>If I create a new project that's in its own Git repository, how do I use JavaScript files from my media repository in my new project in a way that makes it so I don't have to update both copies of the script when I make changes? </p>\n"
        },
        {
            "tags": [
                "botframework",
                "bots",
                "powerbi"
            ],
            "owner": {
                "reputation": 16,
                "user_id": 9645813,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/2a25a0b92d78ebecdefc29651ce35713?s=128&d=identicon&r=PG&f=1",
                "display_name": "J Ackermann",
                "link": "https://stackoverflow.com/users/9645813/j-ackermann"
            },
            "is_answered": false,
            "view_count": 18,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1526463543,
            "creation_date": 1526463543,
            "question_id": 50367346,
            "body_markdown": "Im trying to create a Bot with aid of Microsoft BotFramework which can access information from PowerBI. I found some Blog-posts saying it is not (yet) possible, but i was wondering if there is a way to implement a connection via Microsoft.PowerBI.Api.V2? \r\n\r\nSo far i saw that one can embedd the whole Dashboard in a WebApp with PowerBI.API.V2 but i couldn&#39;t figure out how to access specific information within the dashboard with this API. Is that even possible? \r\n\r\nThanks!",
            "link": "https://stackoverflow.com/questions/50367346/access-powerbi-with-botframework-via-microsoft-powerbi-api-v2",
            "title": "Access PowerBi with BotFramework via Microsoft.PowerBI.Api.V2",
            "body": "<p>Im trying to create a Bot with aid of Microsoft BotFramework which can access information from PowerBI. I found some Blog-posts saying it is not (yet) possible, but i was wondering if there is a way to implement a connection via Microsoft.PowerBI.Api.V2? </p>\n\n<p>So far i saw that one can embedd the whole Dashboard in a WebApp with PowerBI.API.V2 but i couldn't figure out how to access specific information within the dashboard with this API. Is that even possible? </p>\n\n<p>Thanks!</p>\n"
        },
        {
            "tags": [
                "java",
                "groovy",
                "jmeter"
            ],
            "owner": {
                "reputation": 18,
                "user_id": 9685127,
                "user_type": "registered",
                "profile_image": "https://lh3.googleusercontent.com/-XdUIqdMkCWA/AAAAAAAAAAI/AAAAAAAAAAA/4252rscbv5M/photo.jpg?sz=128",
                "display_name": "L&#233;andre",
                "link": "https://stackoverflow.com/users/9685127/l%c3%a9andre"
            },
            "is_answered": false,
            "view_count": 6,
            "closed_date": 1526463644,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1526463542,
            "creation_date": 1526463542,
            "question_id": 50367345,
            "body_markdown": "When I do : `import javax.xml.bind.DatatypeConverter;`\r\n\r\nI get : `unable to resolve class javax.xml.bind.DatatypeConverter`\r\n\r\nI did an update on ubuntu and my groovy script ddon&#39;t work now.\r\n\r\nI think it&#39;s the version of java but i&#39;m not sure",
            "link": "https://stackoverflow.com/questions/50367345/unable-to-resolve-class-javax-xml-bind-datatypeconverter-in-jmeter-groovy",
            "closed_reason": "duplicate",
            "title": "unable to resolve class javax.xml.bind.DatatypeConverter in JMeter/Groovy",
            "body": "<p>When I do : <code>import javax.xml.bind.DatatypeConverter;</code></p>\n\n<p>I get : <code>unable to resolve class javax.xml.bind.DatatypeConverter</code></p>\n\n<p>I did an update on ubuntu and my groovy script ddon't work now.</p>\n\n<p>I think it's the version of java but i'm not sure</p>\n"
        },
        {
            "tags": [
                "python",
                "language-design",
                "least-astonishment"
            ],
            "owner": {
                "reputation": 67951,
                "user_id": 78374,
                "user_type": "registered",
                "accept_rate": 67,
                "profile_image": "https://www.gravatar.com/avatar/f13cb1b1bd8b16bf6dc6eda558e162a0?s=128&d=identicon&r=PG",
                "display_name": "Stefano Borini",
                "link": "https://stackoverflow.com/users/78374/stefano-borini"
            },
            "is_answered": true,
            "view_count": 110061,
            "protected_date": 1326471202,
            "accepted_answer_id": 1145781,
            "answer_count": 30,
            "score": 2003,
            "last_activity_date": 1526463542,
            "creation_date": 1247680837,
            "last_edit_date": 1485649788,
            "question_id": 1132941,
            "body_markdown": "Anyone tinkering with Python long enough has been bitten (or torn to pieces) by the following issue:\r\n\r\n    def foo(a=[]):\r\n        a.append(5)\r\n        return a\r\n\r\n\r\nPython novices would expect this function to always return a list with only one element: `[5]`. The result is instead very different, and very astonishing (for a novice):\r\n\r\n    &gt;&gt;&gt; foo()\r\n    [5]\r\n    &gt;&gt;&gt; foo()\r\n    [5, 5]\r\n    &gt;&gt;&gt; foo()\r\n    [5, 5, 5]\r\n    &gt;&gt;&gt; foo()\r\n    [5, 5, 5, 5]\r\n    &gt;&gt;&gt; foo()\r\n\r\nA manager of mine once had his first encounter with this feature, and called it &quot;a dramatic design flaw&quot; of the language. I replied that the behavior had an underlying explanation, and it is indeed very puzzling and unexpected if you don&#39;t understand the internals. However, I was not able to answer (to myself) the following question: what is the reason for binding the default argument at function definition, and not at function execution? I doubt the experienced behavior has a practical use (who really used static variables in C, without breeding bugs?)\r\n\r\n**Edit**: \r\n\r\nBaczek made an interesting example. Together with most of your comments and Utaal&#39;s in particular, I elaborated further:\r\n\r\n    &gt;&gt;&gt; def a():\r\n    ...     print(&quot;a executed&quot;)\r\n    ...     return []\r\n    ... \r\n    &gt;&gt;&gt;            \r\n    &gt;&gt;&gt; def b(x=a()):\r\n    ...     x.append(5)\r\n    ...     print(x)\r\n    ... \r\n    a executed\r\n    &gt;&gt;&gt; b()\r\n    [5]\r\n    &gt;&gt;&gt; b()\r\n    [5, 5]\r\n\r\nTo me, it seems that the design decision was relative to where to put the scope of parameters: inside the function or &quot;together&quot; with it?\r\n\r\nDoing the binding inside the function would mean that `x` is effectively bound to the specified default when the function is called, not defined, something that would present a deep flaw: the `def` line would be &quot;hybrid&quot; in the sense that part of the binding (of the function object) would happen at definition, and part (assignment of default parameters) at function invocation time.\r\n\r\nThe actual behavior is more consistent: everything of that line gets evaluated when that line is executed, meaning at function definition.",
            "link": "https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument",
            "title": "&quot;Least Astonishment&quot; and the Mutable Default Argument",
            "body": "<p>Anyone tinkering with Python long enough has been bitten (or torn to pieces) by the following issue:</p>\n\n<pre><code>def foo(a=[]):\n    a.append(5)\n    return a\n</code></pre>\n\n<p>Python novices would expect this function to always return a list with only one element: <code>[5]</code>. The result is instead very different, and very astonishing (for a novice):</p>\n\n<pre><code>&gt;&gt;&gt; foo()\n[5]\n&gt;&gt;&gt; foo()\n[5, 5]\n&gt;&gt;&gt; foo()\n[5, 5, 5]\n&gt;&gt;&gt; foo()\n[5, 5, 5, 5]\n&gt;&gt;&gt; foo()\n</code></pre>\n\n<p>A manager of mine once had his first encounter with this feature, and called it \"a dramatic design flaw\" of the language. I replied that the behavior had an underlying explanation, and it is indeed very puzzling and unexpected if you don't understand the internals. However, I was not able to answer (to myself) the following question: what is the reason for binding the default argument at function definition, and not at function execution? I doubt the experienced behavior has a practical use (who really used static variables in C, without breeding bugs?)</p>\n\n<p><strong>Edit</strong>: </p>\n\n<p>Baczek made an interesting example. Together with most of your comments and Utaal's in particular, I elaborated further:</p>\n\n<pre><code>&gt;&gt;&gt; def a():\n...     print(\"a executed\")\n...     return []\n... \n&gt;&gt;&gt;            \n&gt;&gt;&gt; def b(x=a()):\n...     x.append(5)\n...     print(x)\n... \na executed\n&gt;&gt;&gt; b()\n[5]\n&gt;&gt;&gt; b()\n[5, 5]\n</code></pre>\n\n<p>To me, it seems that the design decision was relative to where to put the scope of parameters: inside the function or \"together\" with it?</p>\n\n<p>Doing the binding inside the function would mean that <code>x</code> is effectively bound to the specified default when the function is called, not defined, something that would present a deep flaw: the <code>def</code> line would be \"hybrid\" in the sense that part of the binding (of the function object) would happen at definition, and part (assignment of default parameters) at function invocation time.</p>\n\n<p>The actual behavior is more consistent: everything of that line gets evaluated when that line is executed, meaning at function definition.</p>\n"
        },
        {
            "tags": [
                "excel",
                "vba"
            ],
            "owner": {
                "reputation": 1,
                "user_id": 9797954,
                "user_type": "registered",
                "profile_image": "https://lh3.googleusercontent.com/-dAUenNDFlX0/AAAAAAAAAAI/AAAAAAAAAHg/bgAvolz0Bks/photo.jpg?sz=128",
                "display_name": "Jason He",
                "link": "https://stackoverflow.com/users/9797954/jason-he"
            },
            "is_answered": false,
            "view_count": 22,
            "answer_count": 0,
            "score": -1,
            "last_activity_date": 1526463542,
            "creation_date": 1526447867,
            "last_edit_date": 1526463542,
            "question_id": 50362914,
            "body_markdown": "I know that for pivot table we can create a variable for the command text and connection then use activeworkbooks.pivotcache.connection/commandtext equal those variables. Is there a way to do the same thing with Querytable?\r\n\r\nI tried to do something like `activeworkbooks.worksheets(1).querytables(1).connection`, but it did not work. Also, I have two different connections for two different querytables within the same workbook. \r\n",
            "link": "https://stackoverflow.com/questions/50362914/how-to-enter-and-change-the-connection-string-and-command-text-of-a-querytable-i",
            "title": "How to enter and change the connection string and command text of a Querytable in VBA?",
            "body": "<p>I know that for pivot table we can create a variable for the command text and connection then use activeworkbooks.pivotcache.connection/commandtext equal those variables. Is there a way to do the same thing with Querytable?</p>\n\n<p>I tried to do something like <code>activeworkbooks.worksheets(1).querytables(1).connection</code>, but it did not work. Also, I have two different connections for two different querytables within the same workbook. </p>\n"
        },
        {
            "tags": [
                "javascript",
                "php",
                "html",
                "onclick"
            ],
            "owner": {
                "reputation": 6,
                "user_id": 9316272,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/268a1073e30b267fe27ee6cb3a749813?s=128&d=identicon&r=PG&f=1",
                "display_name": "Mosaab Ahmad",
                "link": "https://stackoverflow.com/users/9316272/mosaab-ahmad"
            },
            "is_answered": false,
            "view_count": 49,
            "answer_count": 1,
            "score": 1,
            "last_activity_date": 1526463532,
            "creation_date": 1526455279,
            "last_edit_date": 1526463532,
            "question_id": 50364650,
            "body_markdown": "I have page php &amp; I have link to download file i need code show img one time\r\nOR onclick Link (Download File) Hide img (nof1.gif) \r\n\r\nNow I am replacing the image(nof1.gif) in an empty img (empty_img.png)\r\nBut when reloading the page the images Again\r\n\r\nThe idea to show the image only once\r\n\r\n&lt;!-- begin snippet: js hide: false console: true babel: false --&gt;\r\n\r\n&lt;!-- language: lang-html --&gt;\r\n\r\n\r\n    &lt;a   href=&quot;&lt;?php echo base_url($l[&#39;admin_file&#39;]); ?&gt;&quot; target=&quot;_blank&quot; &gt;Download File &lt;/a&gt; \r\n\r\n\r\n\r\n    &lt;img id=&quot;my_images&quot; src=&quot;nof1.gif&quot; style=&quot;width:40px;height:30px&quot; \r\n      onclick=&quot;this.src=&#39;empty_img.png&#39;; &quot;\r\n      &gt;\r\n\r\n&lt;!-- end snippet --&gt;\r\n\r\n\r\n\r\nor see the code There is an error and I do not know the solution!!!\r\n\r\n&lt;!-- begin snippet: js hide: false console: true babel: false --&gt;\r\n\r\n&lt;!-- language: lang-html --&gt;\r\n\r\n    &lt;html&gt;\r\n\r\n    &lt;button onclick=&quot;hideImage()&#39;&quot;&gt; Remove img &lt;/button&gt;\r\n    &lt;img id=&quot;my_images&quot; src=&quot;http://karachiairport.com.pk/images/data-section/airline/24-11.png&quot;&gt;\r\n    &lt;script&gt;\r\n\r\n    document.addEventListener(&#39;DOMContentLoaded&#39;, function() {\r\n        // If cookie exist\r\n        if (document.cookie.indexOf(&#39;image_clicked&#39;) &gt; -1) {\r\n            document.getElementById(&quot;my_images&quot;).src=&quot;http://www.1dmag.com/all_asset/img/no-image.png&quot;;\r\n        }\r\n    \t\r\n\r\n        function hideImage() {\r\n            document.getElementById(&quot;my_images&quot;).src=&quot;http://www.1dmag.com/all_asset/img/no-image.png&quot;;\r\n            document.cookie = &quot;image_clicked=true&quot;;\r\n        }\r\n    });\r\n\r\n\r\n    &lt;/script&gt;\r\n\r\n&lt;!-- end snippet --&gt;\r\n\r\n",
            "link": "https://stackoverflow.com/questions/50364650/onclick-link-hide-img-for-ever-same-notification-idea",
            "title": "onclick Link hide img for ever same notification idea",
            "body": "<p>I have page php &amp; I have link to download file i need code show img one time\nOR onclick Link (Download File) Hide img (nof1.gif) </p>\n\n<p>Now I am replacing the image(nof1.gif) in an empty img (empty_img.png)\nBut when reloading the page the images Again</p>\n\n<p>The idea to show the image only once</p>\n\n<p><div class=\"snippet\" data-lang=\"js\" data-hide=\"false\" data-console=\"true\" data-babel=\"false\">\r\n<div class=\"snippet-code\">\r\n<pre class=\"snippet-code-html lang-html prettyprint-override\"><code>&lt;a   href=\"&lt;?php echo base_url($l['admin_file']); ?&gt;\" target=\"_blank\" &gt;Download File &lt;/a&gt; \r\n\r\n\r\n\r\n&lt;img id=\"my_images\" src=\"nof1.gif\" style=\"width:40px;height:30px\" \r\n  onclick=\"this.src='empty_img.png'; \"\r\n  &gt;</code></pre>\r\n</div>\r\n</div>\r\n</p>\n\n<p>or see the code There is an error and I do not know the solution!!!</p>\n\n<p><div class=\"snippet\" data-lang=\"js\" data-hide=\"false\" data-console=\"true\" data-babel=\"false\">\r\n<div class=\"snippet-code\">\r\n<pre class=\"snippet-code-html lang-html prettyprint-override\"><code>&lt;html&gt;\r\n\r\n&lt;button onclick=\"hideImage()'\"&gt; Remove img &lt;/button&gt;\r\n&lt;img id=\"my_images\" src=\"http://karachiairport.com.pk/images/data-section/airline/24-11.png\"&gt;\r\n&lt;script&gt;\r\n\r\ndocument.addEventListener('DOMContentLoaded', function() {\r\n    // If cookie exist\r\n    if (document.cookie.indexOf('image_clicked') &gt; -1) {\r\n        document.getElementById(\"my_images\").src=\"http://www.1dmag.com/all_asset/img/no-image.png\";\r\n    }\r\n\t\r\n\r\n    function hideImage() {\r\n        document.getElementById(\"my_images\").src=\"http://www.1dmag.com/all_asset/img/no-image.png\";\r\n        document.cookie = \"image_clicked=true\";\r\n    }\r\n});\r\n\r\n\r\n&lt;/script&gt;</code></pre>\r\n</div>\r\n</div>\r\n</p>\n"
        },
        {
            "tags": [
                "unit-testing",
                "pycharm"
            ],
            "owner": {
                "reputation": 88,
                "user_id": 3769802,
                "user_type": "registered",
                "accept_rate": 86,
                "profile_image": "https://i.stack.imgur.com/Phk69.jpg?s=128&g=1",
                "display_name": "Sab",
                "link": "https://stackoverflow.com/users/3769802/sab"
            },
            "is_answered": false,
            "view_count": 16,
            "answer_count": 1,
            "score": 2,
            "last_activity_date": 1526463526,
            "creation_date": 1525076839,
            "question_id": 50096652,
            "body_markdown": "When ever I create a new project or open a old one in PyCharm the test runner default to Twisted Trial.\r\n\r\nI did some search and found that I can change project level default by Setting-&gt;Tools-&gt;Python Integrated Tools-&gt; Default test runner\r\n\r\nBut not able to change the global default test runner for PyCharm.\r\n\r\nThe &quot;Tools&quot; option is not available in &quot;Default Settings&quot;.\r\n\r\nHow to set up PyCharm so that the default test runner is &quot;Unittests&quot;.\r\n\r\nPyCharm Version - 2018.1 Community Edition\r\nPython2.7",
            "link": "https://stackoverflow.com/questions/50096652/pycharm-new-project-test-runner-default-to-twisted-trial",
            "title": "PyCharm new project test runner default to Twisted Trial",
            "body": "<p>When ever I create a new project or open a old one in PyCharm the test runner default to Twisted Trial.</p>\n\n<p>I did some search and found that I can change project level default by Setting->Tools->Python Integrated Tools-> Default test runner</p>\n\n<p>But not able to change the global default test runner for PyCharm.</p>\n\n<p>The \"Tools\" option is not available in \"Default Settings\".</p>\n\n<p>How to set up PyCharm so that the default test runner is \"Unittests\".</p>\n\n<p>PyCharm Version - 2018.1 Community Edition\nPython2.7</p>\n"
        },
        {
            "tags": [
                "javascript",
                "canvas",
                "png",
                "base64"
            ],
            "owner": {
                "reputation": 52,
                "user_id": 930473,
                "user_type": "registered",
                "accept_rate": 80,
                "profile_image": "https://www.gravatar.com/avatar/eae9af9664082639299fcf79915b69b7?s=128&d=identicon&r=PG",
                "display_name": "Neuza",
                "link": "https://stackoverflow.com/users/930473/neuza"
            },
            "is_answered": true,
            "view_count": 3090,
            "answer_count": 1,
            "score": 1,
            "last_activity_date": 1526463523,
            "creation_date": 1342715050,
            "question_id": 11565156,
            "body_markdown": "I have some graphics made with canvas that I want to save in a specific folder of the user&#39;s computer as a .png image. \r\n\r\nI&#39;m doing this on the client side:\r\n\r\n \r\n\r\n    $(&quot;.new_canvas&quot;).each( function(index) {\r\n\t\t\t\t\t\r\n\t\t\t\t\tvar dataurl = $(&quot;.new_canvas&quot;)[index].toDataURL(&quot;image/png&quot;);\r\n\t\t\t\t\t   \r\n\t\t\t\t\t$.ajax({\r\n\t\t\t\t\t\t\ttype: &quot;POST&quot;,\r\n\t\t\t\t\t\t\turl: &quot;export_images&quot;,\r\n\t\t\t\t\t\t\tdata: dataurl,\r\n\t\t\t\t\t\t\t}\r\n\t\t\t\t\t});\r\n\t\t\t\t});\r\n\r\nOn the server side I receive the base64 encoded string but how can I save this image in a specific directory? \r\n\r\nI already saw some questions about this and tried a few things but with no success...\r\n\r\nThank you.",
            "link": "https://stackoverflow.com/questions/11565156/save-canvas-as-png-image-in-specific-folder-with-javascript",
            "title": "Save canvas as PNG image in specific folder with javascript",
            "body": "<p>I have some graphics made with canvas that I want to save in a specific folder of the user's computer as a .png image. </p>\n\n<p>I'm doing this on the client side:</p>\n\n<pre><code>$(\".new_canvas\").each( function(index) {\n\n                var dataurl = $(\".new_canvas\")[index].toDataURL(\"image/png\");\n\n                $.ajax({\n                        type: \"POST\",\n                        url: \"export_images\",\n                        data: dataurl,\n                        }\n                });\n            });\n</code></pre>\n\n<p>On the server side I receive the base64 encoded string but how can I save this image in a specific directory? </p>\n\n<p>I already saw some questions about this and tried a few things but with no success...</p>\n\n<p>Thank you.</p>\n"
        },
        {
            "tags": [
                "css",
                "reactjs",
                "webpack",
                "path"
            ],
            "owner": {
                "reputation": 67,
                "user_id": 4290473,
                "user_type": "registered",
                "accept_rate": 40,
                "profile_image": "https://www.gravatar.com/avatar/0cbc2b9b3f95ef8db924e8760d2fa90d?s=128&d=identicon&r=PG&f=1",
                "display_name": "Madhura KM",
                "link": "https://stackoverflow.com/users/4290473/madhura-km"
            },
            "is_answered": false,
            "view_count": 27,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1526463522,
            "creation_date": 1526406888,
            "last_edit_date": 1526408088,
            "question_id": 50356505,
            "body_markdown": "In Webpack, how to configure css loaders to use css files from specific path? I tried below code to fetch files from path &#39;src/styles&#39;, but did not work.\r\n\r\n    rules: [\r\n                    {\r\n                        test: /\\.css$/,\r\n                        use: [\r\n                            &#39;style-loader&#39;,\r\n                            &#39;css-loader&#39;\r\n                        ],\r\n                        include: [\r\n                            path.resolve(__dirname, &quot;src/styles/&quot; ),\r\n                        ]\r\n    \r\n                    }\r\n          ]\r\n\r\n\r\nI have &#39;style.css&#39; under &#39;src/styles&#39; folder. In &#39;index.js&#39; I just did \r\n\r\nimport &#39;style.css&#39;\r\n\r\nbecause I want path to be resolved from loader.\r\nWhen I build, getting below error:\r\n\r\n    **ERROR in ./src/index.js\r\n    Module not found: Error: Can&#39;t resolve &#39;style.css&#39; in &#39;D:\\webpack-demo\\src&#39;\r\n     @ ./src/index.js 2:0-19**\r\n\r\n\r\nPlease help.\r\n",
            "link": "https://stackoverflow.com/questions/50356505/webpack-configure-css-file-paths",
            "title": "Webpack : Configure css file paths",
            "body": "<p>In Webpack, how to configure css loaders to use css files from specific path? I tried below code to fetch files from path 'src/styles', but did not work.</p>\n\n<pre><code>rules: [\n                {\n                    test: /\\.css$/,\n                    use: [\n                        'style-loader',\n                        'css-loader'\n                    ],\n                    include: [\n                        path.resolve(__dirname, \"src/styles/\" ),\n                    ]\n\n                }\n      ]\n</code></pre>\n\n<p>I have 'style.css' under 'src/styles' folder. In 'index.js' I just did </p>\n\n<p>import 'style.css'</p>\n\n<p>because I want path to be resolved from loader.\nWhen I build, getting below error:</p>\n\n<pre><code>**ERROR in ./src/index.js\nModule not found: Error: Can't resolve 'style.css' in 'D:\\webpack-demo\\src'\n @ ./src/index.js 2:0-19**\n</code></pre>\n\n<p>Please help.</p>\n"
        },
        {
            "tags": [
                "c#",
                ".net",
                "oop"
            ],
            "owner": {
                "reputation": 98,
                "user_id": 2168072,
                "user_type": "registered",
                "profile_image": "https://i.stack.imgur.com/vgOw2.png?s=128&g=1",
                "display_name": "Arnulfo Arroyo",
                "link": "https://stackoverflow.com/users/2168072/arnulfo-arroyo"
            },
            "is_answered": true,
            "view_count": 55058,
            "accepted_answer_id": 15400469,
            "answer_count": 6,
            "score": 7,
            "last_activity_date": 1526463519,
            "creation_date": 1363230195,
            "question_id": 15400421,
            "body_markdown": "I am trying to change the text of a Button every time its  clicked.\r\n\r\nButton Starts as &quot;ON&quot;. When I click it the first time it should change to &quot;OFF&quot;, and when I click it again, it should change back to On.\r\n\r\nI understand how to change it to &quot;OFF&quot; when clicked, but I am unsure as to how to check for a secondary click so I can change the text back to &quot;ON&quot;.\r\n\r\nHere is my code so far:\r\n\r\n    private void OrdersButton_Click(object sender, EventArgs e)\r\n    {\r\n           OrdersButton.Text = &quot; Turn Orders Off&quot;;\r\n    }",
            "link": "https://stackoverflow.com/questions/15400421/change-button-text-after-click-then-change-it-back-after-clicking-again",
            "title": "Change button text after click, then change it back after clicking again",
            "body": "<p>I am trying to change the text of a Button every time its  clicked.</p>\n\n<p>Button Starts as \"ON\". When I click it the first time it should change to \"OFF\", and when I click it again, it should change back to On.</p>\n\n<p>I understand how to change it to \"OFF\" when clicked, but I am unsure as to how to check for a secondary click so I can change the text back to \"ON\".</p>\n\n<p>Here is my code so far:</p>\n\n<pre><code>private void OrdersButton_Click(object sender, EventArgs e)\n{\n       OrdersButton.Text = \" Turn Orders Off\";\n}\n</code></pre>\n"
        },
        {
            "tags": [
                "java",
                "jersey",
                "jersey-2.0"
            ],
            "owner": {
                "reputation": 842,
                "user_id": 995514,
                "user_type": "registered",
                "accept_rate": 54,
                "profile_image": "https://www.gravatar.com/avatar/3c27635e9e56d54ec85828df27532654?s=128&d=identicon&r=PG",
                "display_name": "Half_Duplex",
                "link": "https://stackoverflow.com/users/995514/half-duplex"
            },
            "is_answered": true,
            "view_count": 407,
            "answer_count": 2,
            "score": 0,
            "last_activity_date": 1526463516,
            "creation_date": 1490216502,
            "question_id": 42962591,
            "body_markdown": "Is it possible to return a validation annotations message in the response for bad responses? I thought this was possible but I have noticed that our projects are not given detailed bad request messages. \r\n\r\n    @NotNull(message=&quot;idField is required&quot;)\r\n\t@Size(min = 1, max = 15) \r\n\tprivate String idField;\r\n\r\nI&#39;d like to see &quot;idField is required&quot; returned if a request is made that&#39;s missing the idField. I am using jersey 2.0. What I&#39;m seeing for a response is this...\r\n\r\n    {\r\n      &quot;timestamp&quot;: 1490216419752,\r\n      &quot;status&quot;: 400,\r\n      &quot;error&quot;: &quot;Bad Request&quot;,\r\n      &quot;message&quot;: &quot;Bad Request&quot;,\r\n      &quot;path&quot;: &quot;/api/test&quot;\r\n    }",
            "link": "https://stackoverflow.com/questions/42962591/jersey-bean-validation-return-validation-message-for-bad-requests",
            "title": "Jersey bean validation - return validation message for bad requests",
            "body": "<p>Is it possible to return a validation annotations message in the response for bad responses? I thought this was possible but I have noticed that our projects are not given detailed bad request messages. </p>\n\n<pre><code>@NotNull(message=\"idField is required\")\n@Size(min = 1, max = 15) \nprivate String idField;\n</code></pre>\n\n<p>I'd like to see \"idField is required\" returned if a request is made that's missing the idField. I am using jersey 2.0. What I'm seeing for a response is this...</p>\n\n<pre><code>{\n  \"timestamp\": 1490216419752,\n  \"status\": 400,\n  \"error\": \"Bad Request\",\n  \"message\": \"Bad Request\",\n  \"path\": \"/api/test\"\n}\n</code></pre>\n"
        },
        {
            "tags": [
                "ios",
                "iphone",
                "cocoa-touch",
                "uiviewcontroller",
                "ios8"
            ],
            "owner": {
                "reputation": 423,
                "user_id": 2135004,
                "user_type": "registered",
                "accept_rate": 36,
                "profile_image": "https://www.gravatar.com/avatar/ba0a822469cf380e9cfd9b8e2921ff84?s=128&d=identicon&r=PG",
                "display_name": "user2135004",
                "link": "https://stackoverflow.com/users/2135004/user2135004"
            },
            "is_answered": true,
            "view_count": 3248,
            "answer_count": 6,
            "score": 11,
            "last_activity_date": 1526463503,
            "creation_date": 1410930012,
            "question_id": 25882609,
            "body_markdown": "In iOS 7, I was able to handle layout for a custom presented view controller to also account for the in-call status bar by adding constraints where the presented view controller has the same center and the same width and height as the presenting view controller. This was preferred to using auto-resizing for the width and height because, otherwise, when the in-call status bar goes away, the presented view controller would remain pushed down by 20 points from the top for some reason.\r\n\r\nIn iOS 8, however, this trick doesn&#39;t work anymore. For one thing, I found through experimentation that the presenting view controller does not seem to necessarily be in the container view, so I can&#39;t add those constraints to the container view. (The [sample code](https://developer.apple.com/devcenter/download.action?path=/wwdc_2014/wwdc_2014_sample_code/lookinsidepresentationcontrollersadaptivityandcustomanimatorobjects.zip) for the [&quot;A Look Inside Presentation Controllers&quot; 2014 WWDC video](https://developer.apple.com/videos/wwdc/2014/) does not put the presenting view controller in there.) It seems that having the presented view controller filling the container view using either auto-layout or auto-resizing would work, but I found that the container view itself may remain pushed down by 20 points when the in-call status bar disappears (which was not the case in iOS 7).\r\n\r\nI&#39;ve been playing around with the [&quot;A Look Inside Presentation Controllers&quot; sample code](https://developer.apple.com/devcenter/download.action?path=/wwdc_2014/wwdc_2014_sample_code/lookinsidepresentationcontrollersadaptivityandcustomanimatorobjects.zip), but even this code does not handle the in-call status bar correctly. (I&#39;m still trying to get a handle on the new `UIPresentationController` API, by the way.)",
            "link": "https://stackoverflow.com/questions/25882609/how-to-do-layout-to-handle-in-call-double-height-status-bar-for-custom-present",
            "title": "How to do layout to handle in-call (double-height) status bar for custom presented view controller in iOS 8?",
            "body": "<p>In iOS 7, I was able to handle layout for a custom presented view controller to also account for the in-call status bar by adding constraints where the presented view controller has the same center and the same width and height as the presenting view controller. This was preferred to using auto-resizing for the width and height because, otherwise, when the in-call status bar goes away, the presented view controller would remain pushed down by 20 points from the top for some reason.</p>\n\n<p>In iOS 8, however, this trick doesn't work anymore. For one thing, I found through experimentation that the presenting view controller does not seem to necessarily be in the container view, so I can't add those constraints to the container view. (The <a href=\"https://developer.apple.com/devcenter/download.action?path=/wwdc_2014/wwdc_2014_sample_code/lookinsidepresentationcontrollersadaptivityandcustomanimatorobjects.zip\">sample code</a> for the <a href=\"https://developer.apple.com/videos/wwdc/2014/\">\"A Look Inside Presentation Controllers\" 2014 WWDC video</a> does not put the presenting view controller in there.) It seems that having the presented view controller filling the container view using either auto-layout or auto-resizing would work, but I found that the container view itself may remain pushed down by 20 points when the in-call status bar disappears (which was not the case in iOS 7).</p>\n\n<p>I've been playing around with the <a href=\"https://developer.apple.com/devcenter/download.action?path=/wwdc_2014/wwdc_2014_sample_code/lookinsidepresentationcontrollersadaptivityandcustomanimatorobjects.zip\">\"A Look Inside Presentation Controllers\" sample code</a>, but even this code does not handle the in-call status bar correctly. (I'm still trying to get a handle on the new <code>UIPresentationController</code> API, by the way.)</p>\n"
        }
    ],
    "has_more": true,
    "quota_max": 300,
    "quota_remaining": 183
}